toffy/mph_comp.py

```diff
@@ -0,0 +1,162 @@
+    pulse_height_file = fov + '-pulse_height.csv'
+
+    try:
+        median = bin_files.get_median_pulse_height(bin_file_path, fov,
+                                                   target, (mass_start, mass_stop))
+        count_dict = bin_files.get_total_counts(bin_file_path, [fov])
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The FOV name supplied doesn't have a JSON file: {fov}")
+    except ValueError:
+        raise ValueError(f"The target name is invalid: {target}")
+
+    count = count_dict[fov]
+    time = get_estimated_time(bin_file_path, fov)
+
+    out_df = pd.DataFrame({
+        'fov': [fov],
+        'MPH': [median],
+        'total_count': [count],
+        'time': [time]})
+
+    # saves individual .csv  files to bin_file_path
+    if not os.path.exists(os.path.join(csv_dir, pulse_height_file)):
+        out_df.to_csv(os.path.join(csv_dir, pulse_height_file), index=False)
+
+
+def combine_mph_metrics(csv_dir, return_data=False):
+    """Combines data from individual csvs into one
+        Args:
+            csv_dir (str): path where FOV csvs are stored
+            return_data (bool): whether to return dataframe with mph metrics, default False
+
+        Returns:
+            combined mph data for all FOVs
+            """
+
+    # path validation checks
+    io_utils.validate_paths(csv_dir)
+
+    # list bin files in directory
+    fov_files = io_utils.list_files(csv_dir, "-pulse_height.csv")
+
+    # for each csv retrieve mph values
+    combined_rows = []
+    for i, file in enumerate(fov_files):
+        combined_rows.append(pd.read_csv(os.path.join(csv_dir, file)))
+
+    # calculate cumulative sums of total counts and time
+    combined_df = pd.concat(combined_rows)
+    combined_df['cum_total_count'] = combined_df['total_count'].cumsum()
+    combined_df['cum_total_time'] = combined_df['time'].cumsum()
+
+    # save csv to output_dir
+    file_path = os.path.join(csv_dir, 'total_count_vs_mph_data.csv')
+    if os.path.exists(file_path):
+        os.remove(file_path)
+    combined_df.to_csv(file_path, index=False)
+
+    # return data
+    if return_data:
+        return combined_df
+
+
+def visualize_mph(mph_df, regression: bool, save_dir=None):
+    """Create a scatterplot visualizing median pulse heights by FOV cumulative count
+        Args:
+            mph_df (pd.DataFrame): data detailing total counts and pulse heights
+            regression (bool): whether or not to plot regression line
+            save_dir (str): path of directory to save plot to
+            """
+
+    # path validation checks
+    if save_dir is not None:
+        io_utils.validate_paths(save_dir)
+
+    # visualize the median pulse heights
+    plt.style.use('dark_background')
+    # plt.title('FOV total counts vs median pulse height')
+    fig = plt.figure()
+    ax1 = fig.add_subplot(111)
+    ax2 = ax1.twiny()
+    x = mph_df['cum_total_count']
+    y = mph_df['MPH']
+    x_alt = mph_df['cum_total_time']
```

Comment:
This should be in hours so that it's interpretable. 400k milliseconds is not super intuitive